### PR TITLE
check workers started stat for envoy readiness

### PR DESCRIPTION
**Pull request by ramaraochavali**
Envoy has added a `workers_started` stat that indicates all listeners are bound to sockets and hence it is fully ready to take traffic. This PR enhances the readiness probe to use that stat.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn PTAL

File: pilot/cmd/pilot-agent/status/util/stats.go [hunk 1]
```diff
@@ -58,30 +61,35 @@ func (s *Stats) String() string {
...
 		{name: statServerState, value: &s.ServerState},
+		{name: statWorkersStarted, value: &s.WorkersStarted},
 	}
 	if err := parseStats(stats, allStats); err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
-	return &s.ServerState, nil
+	return &s.ServerState, s.WorkersStarted, nil
```
Comment (howardjohn) [inline pilot/cmd/pilot-agent/status/util/stats.go]: is this the number of workers started?

File: pilot/cmd/pilot-agent/status/ready/probe.go [hunk 2]
```diff
@@ -72,5 +72,9 @@ func (p *Probe) checkServerState() error {
 		return fmt.Errorf("server is not live, current state is: %v", admin.ServerInfo_State(*state).String())
 	}
 
+	if ws != 1 {
```
Comment (howardjohn) [inline pilot/cmd/pilot-agent/status/ready/probe.go]: is this always 0 or 1? if so why not make this a bool here.

File: pilot/cmd/pilot-agent/status/ready/probe.go [hunk 3]
```diff
@@ -72,5 +72,9 @@ func (p *Probe) checkServerState() error {
...
+		return fmt.Errorf("workers have not yet started, %d", ws)
```
Comment (howardjohn) [inline pilot/cmd/pilot-agent/status/ready/probe.go]: nit - the ", 0" part is confusing

File: pilot/cmd/pilot-agent/status/util/stats.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/status/util/stats.go]: no. It is just a gauge that indicates whether all workers have started or not

File: pilot/cmd/pilot-agent/status/ready/probe.go (hunk 2 above)
Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/status/ready/probe.go]: Yes. It is always zero or 1. I can make it bool.

**Pull request merged**